### PR TITLE
[READY] Twohanded component port

### PR DESCRIPTION
**Pull request by covertcorvid**
## About The Pull Request
I'm back with this again.
Ports https://github.com/tgstation/tgstation/pull/49422 plus using signals to get rid of the wielded var because I don't like duplicating information

Uses named arguments, so requires #3883 to be merged first

## Why It's Good For The Game
Makes it possible to have twohanded behavior on other item paths

## Changelog
:cl:
code: Added two_handed component
refactor: Updated all existing two handed items to use the new component
/:cl:

### CODE REVIEW

File: code/__DEFINES/components.dm [hunk 1]
```diff
@@ -394,4 +408,4 @@
 #define COMSIG_XENO_MONKEY_CLICK_CTRL "xeno_monkey_click_ctrl"				//! from monkey CtrlClickOn(): (/mob)
 
 //Heretics stuff
-#define COMSIG_HUMAN_VOID_MASK_ACT "void_mask_act"
\ No newline at end of file
+#define COMSIG_HUMAN_VOID_MASK_ACT "void_mask_act"
```
Comment (Archanial) [inline code/__DEFINES/components.dm]: we don't have void path

Comment (covertcorvid) [reply code/__DEFINES/components.dm]: This signal is from the Mask of Madness port https://github.com/BeeStation/BeeStation-Hornet/pull/3446
The editor just automatically added a newline to the end of the file.

**Comment by covertcorvid** [comment]
For the record the map changes are 99% just kirbyplants

**Comment by ike709** [comment]
Christ I hate the scope of this. Fix conflicts and ping for a testmerge.

**Comment by PowerfulBacon** [comment]
Block power wielded and unwielded seems to not he accounted for for some items such as the bo staff

**Comment by covertcorvid** [comment]
> 
> 
> Block power wielded and unwielded seems to not he accounted for for some items such as the bo staff

Thanks, good eye! I did another pass on all affected items that use wielded block power and this should be fixed.

**Review by PowerfulBacon — DISMISSED** [review]

**Comment by PowerfulBacon** [comment]
honestly would have been easier to keep the twohanded typepath and not deal with the 2.4 billion merge conflicts lol

**Comment by PowerfulBacon** [comment]
Does adding multiple of the same component to an object override the old ones?

**Comment by covertcorvid** [comment]
> 
> 
> Does adding multiple of the same component to an object override the old ones?

Adding the component to something with an existing twohanded component will copy over all the unique arguments, but anything that got redefined in the new component will override the old one. Similar to object inheritance.

**Comment by covertcorvid** [comment]
Unfortunate that it gets rid of the approval when I clear conflicts lol

**Review by utilbird — DISMISSED** [review]

File: code/modules/mining/equipment/kinetic_crusher.dm [hunk 2]
```diff
@@ -30,23 +28,27 @@
 	var/light_on = FALSE
 	var/brightness_on = 5
 
-/obj/item/twohanded/kinetic_crusher/Initialize()
+/obj/item/kinetic_crusher/Initialize()
+	. = ..()
```
Comment (PowerfulBacon) [inline code/modules/mining/equipment/kinetic_crusher.dm]: Completely pointless initialize proc

**Review by PowerfulBacon — COMMENTED** [review]
We should probably testmerge this first, would have been easier keeping the twohanded typepath but hopefully it doesn't conflict again.

**Comment by covertcorvid** [comment]
> 
> 
> We should probably testmerge this first, would have been easier keeping the twohanded typepath but hopefully it doesn't conflict again.

The TG guy tried to do that and people there said "no bad, if you're doing it do it all the way" so *shrug*

**Review by utilbird — APPROVED** [review]

**Review by PowerfulBacon — APPROVED** [review]

**Comment by Martinpachu** [comment]
all two handed weapons are broken help

**Comment by covertcorvid** [comment]
What.
I will look into this in like an hour

**Comment by Martinpachu** [comment]
they turn into ''nosprite'',i experienced it with the PKC after wielding it and apparently its with all two handed

**Comment by covertcorvid** [comment]
> they turn into ''nosprite'',i experienced it with the PKC after wielding it and apparently its with all two handed

Fucks sake. I did some tests but apparently wasn't exhaustive enough. Thanks for letting me know

**Pull request merged**